docs/cache-and-resume.md

```diff
@@ -0,0 +1,198 @@
+(cache-resume-page)=
+
+# Caching and resuming
+
+One of the core features of Nextflow is the ability to cache task executions and re-use them in subsequent runs to minimize duplicate work. Resumability is useful both for recovering from errors and for iteratively developing a pipeline. It is similar to [checkpointing](https://en.wikipedia.org/wiki/Application_checkpointing), a common practice used by HPC applications.
+
+You can enable resumability in Nextflow with the `-resume` flag when launching a pipeline with `nextflow run`. In most cases, that is all you need to do and resumability will "just work". This page describes Nextflow's caching behavior in more detail in order to help advanced users understand how the cache works and troubleshoot it when it doesn't work.
```

Comment:
I wonder if we add some reference to the `Resuming from a specific run` section, for anyone skimming and not scrolling. Many people do not realise that `-resume` can take an argument.

Reply:
If you have a place in mind, feel free to suggest it. I couldn't think of any. I don't think it needs to be linked here because it's on the same page

Reply:
Fair enough 👍🏻 Yes nothing obvious springs to mind either, without being very duplicative.